### PR TITLE
Add overview of literals

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

**Review by mverzilli — APPROVED** [review]

File: syntax_and_semantics/literals.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+| [Bool](./literals/bool.html)              | `true`, `false`                             |
+| [Integers](./literals/integers.html)      | `18`, `-12`, `19_i64`, `14_u32`,`64_u8`     |
+| [Floats](./literals/floats.html)          | `1.0`, `1.0_f32`, `1e10`, `-0.5`            |
+| [Char](./literals/char.html)              | `'a'`, `'\n'`, `'あ'`                       |
+| [String](./literals/char.html)            | `"foo\tbar"`, `%("あ")`, `%q(foo #{foo})`   |
+| [Symbol](./literals/symbol.html)          | `:symbol`, `:"foo bar"`                     |
+| [Array](./literals/array.html)            | `[1, 2, 3]`, `[1, 2, 3] of Number`, `%w(one two three)` |
+| [Array-like](./literals/array.html)       | `Set{1, 2, 3}`                              |
+| [Hash](./literals/hash.html)              | `{"foo" => 2}`, `{} of String => Int32`     |
+| [Hash-like](./literals/hash.html)         | `MyType{"foo" => "bar"}`                    |
```
Comment (RX14) [inline syntax_and_semantics/literals.md]: Can we link to sections here?

**Review by RX14 — CHANGES_REQUESTED** [review]

File: syntax_and_semantics/literals.md (hunk 1 above)
Comment (straight-shoota) [reply syntax_and_semantics/literals.md]: Great idea!

File: syntax_and_semantics/literals.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+| Literal                                   | Sample values                               |
+|---                                        |---                                          |
+| [Nil](./literals/nil.html)                | `nil`                                       |
+| [Bool](./literals/bool.html)              | `true`, `false`                             |
+| [Integers](./literals/integers.html)      | `18`, `-12`, `19_i64`, `14_u32`,`64_u8`     |
+| [Floats](./literals/floats.html)          | `1.0`, `1.0_f32`, `1e10`, `-0.5`            |
+| [Char](./literals/char.html)              | `'a'`, `'\n'`, `'あ'`                       |
+| [String](./literals/char.html)            | `"foo\tbar"`, `%("あ")`, `%q(foo #{foo})`   |
+| [Symbol](./literals/symbol.html)          | `:symbol`, `:"foo bar"`                     |
+| [Array](./literals/array.html)            | `[1, 2, 3]`, `[1, 2, 3] of Number`, `%w(one two three)` |
```
Comment (asterite) [inline syntax_and_semantics/literals.md]: `[] of Number` doesn't compile

**Review by asterite — CHANGES_REQUESTED** [review]

**Review by RX14 — APPROVED** [review]

**Comment by mverzilli** [comment]
Thank you!

**Pull request merged**